pkg/datapath/iptables/reconciler.go

```diff
@@ -256,12 +256,15 @@ stop:
 				health.OK("iptables rules full reconciliation completed")
 				firstInit = false
 				stateChanged = false
-				// close all channels waiting for reconciliation
-				for _, ch := range updatedChs {
-					close(ch)
-				}
-				updatedChs = updatedChs[:0]
 			}
+
+			// close all channels waiting for reconciliation
+			// do this even in case of a failed reconciliation, to avoid
+			// blocking consumer goroutines indefinitely.
+			for _, ch := range updatedChs {
+				close(ch)
```

Comment:
Would it be better for the waiters to get the error value in the channel in case reconciliation failed?

Reply:
With the current model I'm not sure that would be much useful: the reconciler keeps retrying the operations until success, so the information in the channel would describe the result of the first attempt only. The consumer won't have a chance to know what happens later (either persistent failure or successful recovery).
But yeah, we might change this in the future returning a channel that pumps each reconciling error and closes only after a success, if needed.